c7n/resources/iam.py

```diff
@@ -37,10 +37,243 @@ class Role(QueryResourceManager):
+        results = []
+        client = local_session(self.manager.session_factory).client('lambda')
+        fxs = client.list_functions()['Functions']
+        for fx in fxs:
+            if 'Role' not in fx:
+                continue
+            results.append(fx['Role'])
+        return results
+
+    def scan_ecs_roles(self):
+        results = []
+        client = local_session(self.manager.session_factory).client('ecs')
+        for cluster in client.describe_clusters()['clusters']:
+            svcs = client.list_services(cluster=cluster)['serviceArns']
+            for svc in client.describe_services(cluster=cluster,services=svcs)['services']:
+                if 'roleArn' not in svc:
+                    continue
+                results.append(svc['roleArn'])
+        return results
+
+    def scan_asg_roles(self, rolearn):
+        results = []
+        client = local_session(self.manager.session_factory).client('autoscaling')
+        asgs = client.describe_launch_configurations()['LaunchConfigurations']
+        for asg in asgs:
+            if 'IamInstanceProfile' not in asg:
+                continue
+            if rolearn:
+                for arn in self.get_profile_role_arn(asg['IamInstanceProfile']):
+                    results.append(arn)
+            else:
+                results.append(asg['IamInstanceProfile'])
+        return results
+
+    def scan_ec2_roles(self, rolearn):
+        results = []
+        client = local_session(self.manager.session_factory).client('ec2')
+        instances = client.describe_instances()['Reservations']
+        for instance in instances:
+            for ec2 in instance['Instances']:
+                if 'IamInstanceProfile' not in ec2:
+                    continue
+                if rolearn:
+                    for arn in self.get_profile_role_arn(ec2['IamInstanceProfile']['Arn']):
+                        results.append(arn)
+                else:
+                    results.append(ec2['IamInstanceProfile']['Arn'])
+        return results
+
+
+###################
+#    IAM Roles    #
+###################
+
+
+@Role.filter_registry.register('inuse')
+class InUseIamRole(IamRoleUsage):
+
+    schema = type_schema('inuse')
+
+    def process(self, resources, event=None):
+        roles = self.get_service_roles()
+        results = []
+        for r in resources:
+            if r['Arn'] in roles:
+                results.append(r)
+        return results
+
+
+@Role.filter_registry.register('unused')
+class UnusedIamRole(IamRoleUsage):
+
+    schema = type_schema('unused')
+
+    def process(self, resources, event=None):
+        roles = self.get_service_roles()
+        results = []
+        for r in resources:
+            if r['Arn'] not in roles:
+                results.append(r)
+        return results
+
+
+######################
+#    IAM Policies    #
+######################
+
+
+@Policy.filter_registry.register('attached')
+class IamAttachedPolicies(ValueFilter):
```

Comment:
don't subclass value filter if not using any of the logic from it, use the Filter base class. Please review all subclassing of value filter in this PR, this seems to be a consistent issue.